### DISCUSSION TITLE
How to break down multi meta variable and rewrite each of them separately

### DISCUSSION THREAD

**Discussion by yidingww** [ast-grep/ast-grep#488] [Q&A]
For example, if I want to match:
```ts
import {merge, omit} from 'lodash';
```

and rewrite to:
```ts
import merge from 'lodash/merge';
import omit from 'lodash/omit';
```

How should I do it? If I use `import { $$$LODASH_PKGS } from 'lodash'` as the pattern and `import $$$LODASH_PKGS from 'lodash/$$$LODASH_PKGS'` as the fix, end end result will become `import merge, omit from 'lodash/merge, omit'`, which is **invalid syntax**

**Comment by HerringtonDarkholme** [comment]
It is not possible now doing it in YAML now since it requires mapping a multi-variable match to multiple terms.

You can do it using napi instead.  https://ast-grep.github.io/guide/api-usage.html
